src/main/java/de/craften/ui/swingmaterial/MaterialFrameWrapper.java

```diff
@@ -0,0 +1,372 @@
+//project I worked on some time ago.
+public class MaterialFrameWrapper {
+    /** The frame this object wraps around. */
+    private MaterialFrame frame;
+    private MaterialTitleBar titleBar;
+
+    public MaterialFrameWrapper(MaterialFrame frame) {
+        this.frame = frame;
+        titleBar = new MaterialTitleBar(frame);
+    }
+    
+    /**
+     * Wraps the decorators bundled with this {@code MaterialFrameWrapper}
+     * around the target frame's container panel. This is called by the frame
+     * once its container panel is set and in place.
+     * @param contentPane 
+     */
+    public void wrapAround(Container contentPane) {
+        Container root = contentPane.getParent();
+        root.add(titleBar);
+        titleBar.setVisible(true);
+    }
+    
+    /**
+     * Sets the default color of this wrapper. This affects the decorations
+     * bundled with it, like the title bar and control buttons.
+     * @param color the new color of this wrapper.
+     * @see MaterialFrame#setColor(java.awt.Color)
+     */
+    public void setColor(Color color) {
+        titleBar.setBackground(color);
+    }
+    
+    /**
+     * Gets the default color of this wrapper. More precisely, the color of the
+     * decorations bundled with it, like the title bar and control buttons.
+     * @return the current color of this wrapper.
+     * @see MaterialFrame#getColor()
+     */
+    public Color getColor() {
+        return titleBar.getBackground();
+    }
+    
+    public void updateTitle() {
+        titleBar.setTitle(frame.getTitle());
+    }
+    
+    /**
+     * Causes this wrapper to lay out its components around the frame's
+     * container pane. It is called from {@link MaterialFrame#doLayout()}, which
+     * is triggered each time the frame is resized. You can also request the
+     * layout to be done by calling {@link JFrame#validate()}.
+     * @see JFrame#validate
+     */
+    public void doLayout() {
+        titleBar.setBounds(0, 0, frame.getWindowWidth(), titleBar.getHeight());
+        titleBar.doLayout();
+        frame.getContentPane().setBounds(0, titleBar.getHeight(),
+                frame.getWindowWidth(),
+                frame.getWindowHeight() - titleBar.getHeight());
+    }
+    
+    protected static class MaterialTitleBar extends JPanel
+                implements ActionListener {
+        private FrameControler control;
+        private MaterialFrame frame;
+        
+        private MaterialButton btnClose;
+        private MaterialButton btnMinimize;
+        private MaterialButton btnMaximize;
+        private MaterialButton btnRestore;
+        
+        private JLabel titleLabel;
+        
+        private boolean ready = false;
+        
+        protected MaterialTitleBar(MaterialFrame frame) {
+            this.frame = frame;
+            control = new FrameControler(frame);
+            Font font = MaterialIcons.ICON_FONT.deriveFont(20f);
+            btnClose = new MaterialButton();

```

Comment:
Sure, go ahead. Maybe you could try adding that functionality directly into `MaterialIconButton`.